homework_3/solution.js

```diff
@@ -1,6 +1,43 @@
-var sum = 0;
+var colors = require("colors");
 
-for (var i = 2; i < process.argv.length; i++) {
-  sum += Number(process.argv[i]);
+var mathFuns = {
+    "+" : function(x, y) { return x + y; },
+    "-" : function(x, y) { return x - y; },
+    "x" : function(x, y) { return x * y; },
+    "/" : function(x, y) { return x / y; },
+    "^" : function(x, y) { return Math.pow(x, y); }
+}
+
+function isNumber(n) {
+  return !isNaN(parseFloat(n)) && isFinite(n);
+}
+
+module.exports = function (equation) {
+    var ops = equation.split(" ");
+    var result;
+
+    if (ops.length < 2) {
+        result = Number(ops[0]);
```

Comment:
Since you're calling `Number(ops[0])` twice, you could maybe declare `curVal` at the top of you the function and assign it to `Number(ops[0])` immediately. Then this line could just be `result = curVal;`.